### DISCUSSION TITLE
Support custom aggregated queries through the UI

### DISCUSSION THREAD

**Discussion by emieldatalytica** [langfuse/langfuse#4000] [Ideas]
### Describe the feature or potential improvement

Suppose I want to answer a question like the following: 
> What is the average score X of traces with metadata Y = "Z"?

Currently, the process to analyze aggregated data points would involve exporting the data, and then using external tools for further analysis. 

A feature within Langfuse that allows users to perform custom aggregated queries directly through the UI would empower users to quickly generate insights directly within Langfuse and enable dynamic calculations of averages, sums, or other statistics based on specific conditions. 

In short, I'm requesting the following:

- **Query Builder Interface**: A user-friendly query builder that allows users to specify conditions and aggregation functions (like AVG, SUM, COUNT).

Ideally, there would also be an integration with the dashboard to visualize this data (and update these queries in real-time)
